source/material_model/utilities.cc

```diff
@@ -618,13 +683,34 @@ namespace aspect
               if (in.eof())
                 break;
 
-              density_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[0]];
-              thermal_expansivity_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[1]];
-              specific_heat_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[2]];
-              vp_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[3]];
-              vs_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[4]];
-              enthalpy_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[5]];
-
+              if (first_natural_variable == "T(K)")
+                {
+                  density_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[0]];
+                  thermal_expansivity_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[1]];
+                  specific_heat_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[2]];
+                  vp_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[3]];
+                  vs_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[4]];
+                  enthalpy_values[i%n_temperature][i/n_temperature]=row_values[prp_indices[5]];
+
+                  for (unsigned int n=0; n<phase_volume_fractions.size(); n++)
+                    {
+                      phase_volume_fractions[n][i%n_temperature][i/n_temperature]=row_values[phase_column_indices[n]];
+                    }
+                }
+              else // first_natural_variable == "P(bar)"
```

Comment:
Wait, was this here before? What does it do? Is that change related to the other changes in this pull request?

Reply:
No, it wasn't here before - this was the headache I found on Wednesday night.

The ordering of the first two columns in the PerpleX table files dictates whether the inner loop is over temperature or pressure. The first column is always the inner loop.

I've now added some documentation to describe this. The existing tests cover the possible range of possibilities.

Reply:
I am just saying: this is independent of the phase stuff, right? Before, we just didn't allow having pressure as the inner loop? We could also just make sure temperature is always the first column. Is there a reason why one would want to have pressure first?

Reply:
But I guess since it's already part of this pull request, and you've made a test for it, if there's a reason why someone would want to have pressure first, let's just leave this as it is.

Reply:
Thanks; yes, I should have split this into a separate PR, but now that it's in there...

The reason why I think this is needed is that I genuinely don't understand how PerpleX chooses how to output the table. The ordering of P-first or T-first seems to have changed a few months ago, or maybe there's a flag that I don't know about.

Either way, if I don't know about how PerpleX makes these choices, I'm sure there are other users who would be similarly PerpleXed if their completely-reasonable-looking input file produced garbage. It also makes the code somewhat more future-proof.